leader-election-impl/src/test/java/com/palantir/leader/proxy/AwaitingLeadershipProxyTest.java

```diff
@@ -132,6 +133,49 @@ public void shouldNotMapInterruptedExceptionToNCLEIfLeadingStatusDoesNotChange()
                 .hasMessage(TEST_MESSAGE);
     }
 
+    @Test
+    public void shouldGainLeadershipImmediatelyIfAlreadyLeading() throws Exception {
+        when(leaderElectionService.blockOnBecomingLeader()).then(invocation -> {
+            Uninterruptibles.sleepUninterruptibly(250, TimeUnit.MILLISECONDS);
+            return leadershipToken;
+        });
```

Comment:
nit: do we need this? If I'm understanding things correctly we don't want this to be called by our thread, but we don't really have a way of checking that in any case.